### DISCUSSION TITLE
How does openfast consider aerodynamic damping

### DISCUSSION THREAD

**Discussion by Vain-TLD** [OpenFAST/openfast#1866] [Q&A]
I have focused on the https://forums.nrel.gov/t/aerodynamic-damping-of-the-nacelle-tower-and-the-blades/3889 website for the discussion of the aerodynamic damping. But I still can't understand how structural module and the aerodynamic module fit together.My question is how does the motion of the blade affect the aerodynamic load?Where can I find references to better understand how openfast addresses this issue?

**Comment by jjonkman** [comment]
Dear @Vain-TLD,

The forum post you linked to summarizes the basics--the aerodynamic loads calculated by AeroDyn and sent to structural module depend on the structural motions (position/orientation and velocity of the various analysis nodes) sent from the structural module to AeroDyn.  Basically, the relative velocity between the wind and structure is used rather than the pure wind velocity in most terms used by AeroDyn.  I would suggest referring to the AeroDyn documentation and associated papers for more information on how AeroDyn computes aerodynamics loads; https://openfast.readthedocs.io/en/main/source/user/aerodyn/index.html.

Best regards,

**Reply by Vain-TLD** [reply]
Dear Dr. Jonkman,

Thank you for your prompt response and assistance. I have reviewed the materials on the website https://openfast.readthedocs.io/en/main/source/user/aerodyn/index.html and have been examining the source codes for BEMT and DBEMT. However, I still have the following three areas of confusion that I hope you can assist me with:

1. In the document "Dynamic inflow and unsteady aerodynamics models for modal and stability analyses in OpenFAST," I found information about the impact of blade motion on aerodynamics. Yet, in studying the BEMTUnCoupled.f90 file, it appears that blade motion isn't accounted for. Does this imply that when selecting "WakeMod - Type of wake/induction model (switch) {0=none, 1=BEMT, 2=DBEMT, 3=OLAF} [WakeMod cannot be 2 or 3 when linearizing]," **choosing 1 (BEMT) neglects the aerodynamic damping effect, while choosing 2 (DBEMT) does consider aerodynamic damping?**

2.I've recently read several documents regarding aerodynamic damping:
a. "Passive and Active Vibration Control of Renewable Energy Structures" (Doctoral thesis)
b. "Aerodynamic Damping and Seismic Response of Horizontal Axis Wind Turbine Towers"
c. "Effects of aerodynamic damping on the tower load of offshore horizontal axis wind turbines"
![BEMT](https://github.com/OpenFAST/openfast/assets/127289637/0e79dcd2-45ad-4931-a908-3ae94dc74f89)

These documents appear to reflect aerodynamic damping by considering blade motion. Specifically, they mention：
V1=（1-a）V0-u1
V2=（1+a'）Ωr-u2
(Similar definitions are found in the AeroDyn Theory Manual, where u1=ve-op,u2=ve-ip)
![BEMT2](https://github.com/OpenFAST/openfast/assets/127289637/d8a8b626-dc45-4bfe-b3af-b35644b733cc)
However, in BEMT.f90, within the function ComputePhiWithInduction, the calculation of inflow angle seems somewhat different:
x = (Vx * cos(cantAngle) + xVelCorr) * (1.0_R8Ki - a)
y = Vy * (1.0_ReKi + aprime)
**I am puzzled about whether Vx and Vy in BEMT.f90 already account for blade motion. Does this mean that u1 and u2 should also consider the induction factors, such as V1 = (1 - a) * (V0 - u1) and V2 = (1 + a') * (Ωr - u2)?**

3. I have another hypothesis that I'm not sure is correct. During the BEMT iteration to solve for a and a', what if we don't consider the influence of u1 and u2 (Inflow Angle = arctan((1 - a) * V0 / (1 + a') * Ωr)) during the iteration process? Once the iteration is completed and we have obtained a and a', we can then calculate Inflow Angle as Inflow Angle = arctan((1 - a) * V0 - u1) / ((1 + a') * Ωr - u2), as well as cl, cd, and aerodynamic forces.

I would greatly appreciate your insights and guidance on these matters.

Thank you for your time and assistance.

Best regards,
Vain_TLD
